x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -601,14 +614,16 @@ export function App({
                 data.query.timefilter.timefilter.setTime(dateRange);
                 trackUiEvent('app_date_change');
               } else {
+                // Query has changed, renew the session id.
+                // Time change will be picked up by the time subscription
+                setState((s) => ({
+                  ...s,
+                  searchSessionId: data.search.session.start(),
+                }));
                 trackUiEvent('app_query_change');
               }
               setState((s) => ({
                 ...s,
-                dateRange: {
-                  fromDate: dateRange.from,
-                  toDate: dateRange.to,
-                },
```

Comment:
So we're using the `timeFilter` manager instead now?

Reply:
The code was already using the `timeFilter`, but this code here was covering the scenario where the data didn't change but we want the state to trigger a new render ("Refresh" click).
Previously when the `dateRange` values changed the state was update twice with the same content, triggering a double rendering. Now this scenario is covered by the `timeFilter` subscription.
In case of no `dateRange` change ("Refresh" button), the session id is regenerated triggering a re-render now.

We still need the `dateRange` state, but we can reduce the number of time we update it.

Reply:
Discussed with @flash1293 and found a simpler solution to that